src/views/arrange/index.js

```diff
@@ -62,20 +56,81 @@ export class Arrange extends Component {
           ...response.data.arrangement,
           unsnapshot_items: items
         }
-        this.setState(stateVal)
+        this.props.setRealData(stateVal)
         return Promise.resolve();
       })
       .catch(err => {
         return Promise.reject(err)
       })
+    // For test without API call  
+    /* this.setState(
+      {
+        "_id": "aJDUX35L6",
+        "containers": [
+          {
+          "_id": "cBRXCRDX0",
+          "name": "chia van",
+          "size": 8
+          },
+          {
+          "_id": "cF7X2WFXW",
+          "name": "nathan car",
+          "size": 8
+          }
+        ],
+        "is_deleted": false,
+        "items": [
+          {
+          "_id": "i9LJ1YT7H",
+          "name": "gideon",
+          "size": 1
+          },
+          {
+          "_id": "iBCMM3B14",
+          "name": "gideon luggage",
+          "size": 1
+          },
+          {
+          "_id": "iRREMT1HT",
+          "name": "jeff",
+          "size": 1
+          },
+          {
+          "_id": "iO0SQPBSV",
+          "name": "nathan",
+          "size": 1
+          },
+          {
+          "_id": "iPOWL7Z7F",
+          "name": "moses",
+          "size": 1
+          }
+        ],
+        "name": "first arrangement",
+        "snapshots": [
+          {
+            "_id": "sNC096STL",
+            "name": "only snapshot",
+            "snapshot": {
+              "cBRXCRDX0": [
+                "iO0SQPBSV",
+                "iBCMM3B14",
+                "iPOWL7Z7F"
+              ]
+            }
+          }
+        ],
+        "timestamp": 1538582360.173882
+      }
+    ) */
   }
 
   render () {
     return (
       <Grid container spacing={24} className="arrange">
         <Grid item xs={12} sm={4}>
           <Typography variant="headline" gutterBottom align="left">
-            {this.state.name}
+            {this.props.real.name}

```

Comment:
What is "real"? I have no idea what that is haha...

Reply:
real is the new item, that we input into the React. 
it differentiate the JSON object, because it does not update the server JSON data  yet.